GetStream/stream-chat-android
# [4664] Expose channel options state inside `SelectedChannelMenu`

### 🎯 Goal

closes #4664

### 🛠 Implementation details

Exposed channel options state inside `SelectedChannelMenu`

### 🧪 Testing

Test A - Compare with old implementation:

1. Launch the Compose app on this branch, open up options for a few channels in which the user has different roles (user, owner, etc)
2. Long press on the channel item until the `SelectedChannelMenu` pops up
3. Take a screenshot to help you remember the available options
4. Cross check with the `v5` branch to make sure the options are identica

Test B -Modify channel options

<details>

<summary>Patch which modifies channel options</summary>

```
Index: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/ChannelsScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/ChannelsScreen.kt b/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/ChannelsScreen.kt
--- a/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/ChannelsScreen.kt	(revision 1e3aba6abe44345c9bd84558813c740af0a7f8f5)
+++ b/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/ChannelsScreen.kt	(date 1676372618126)
@@ -59,6 +59,7 @@
 import io.getstream.chat.android.compose.ui.channels.list.ChannelList
 import io.getstream.chat.android.compose.ui.components.SearchInput
 import io.getstream.chat.android.compose.ui.components.SimpleDialog
+import io.getstream.chat.android.compose.ui.components.messageoptions.defaultMessageOptionsState
 import io.getstream.chat.android.compose.ui.theme.ChatTheme
 import io.getstream.chat.android.compose.viewmodel.channels.ChannelListViewModel
 import io.getstream.chat.android.compose.viewmodel.channels.ChannelViewModelFactory
@@ -203,7 +204,8 @@
                         else -> listViewModel.performChannelAction(action)
                     }
                 },
-                onDismiss = { listViewModel.dismissChannelAction() }
+                onDismiss = { listViewModel.dismissChannelAction() },
+                channelOptions = listOf()
             )
         }
 
```

</details>

1. Apply the patch
2. Run the app from this branch
3. Open the menu by long pressing on a channel item
4. Validate that options are now empty 

Test C - Run docusaurus:

1. Run docusaurus
2. I don't think I've mentioned you should run docusaurus yet, so run docusaurus if you haven't ran docusaurus
3. Check the new additions in the `SelectedChannelMenu` page

### ☑️Contributor Checklist

#### General
- [x] PR targets the ~~`develop`~~ `v5` branch

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media2.giphy.com/media/DIAMLpSqEwJoY/giphy.gif?cid=790b76110cc86e1f41f1daa392b62870c4097ffb53e99b79&rid=giphy.gif&ct=g)
